hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/AbstractRecordStore.java

```diff
@@ -206,13 +206,13 @@ protected Record putNewRecord(Data key, Object value, long ttlMillis,
         return record;
     }
 
-    protected Object runMapStore(Record record, Data key, Object value, long now, UUID transactionId) {
+    protected Object runMapStore(Record record, Data key, Object newValue, long now, UUID transactionId) {
         long expirationTime = record.getExpirationTime();
-        value = mapDataStore.add(key, value, expirationTime, now, transactionId);
+        newValue = mapDataStore.add(key, newValue, expirationTime, now, transactionId);
         if (mapDataStore.isPostProcessingMapStore()) {
-            recordFactory.setValue(record, value);
+            storage.updateRecordValue(key, record, newValue);
```

Comment:
This looks bad - seems that, if there was a post-processing map store, we previously didn't put the new value into hot restart, we didn't dispose of the old value and we didn't recalculate the cost estimate. Is this still the case in 3.12.x?
On another note, the `RecordFactory#setValue` method now looks used only in one place which might be replaced.